node/silkworm/downloader/internals/working_chain.cpp

```diff
@@ -508,7 +508,7 @@ void WorkingChain::connect(Segment::Slice segment_slice) {  // throw segment_cut
     if (!attachment_link)
         throw segment_cut_and_paste_error("segment cut&paste error, connect attachment link not found for " +
                                           to_hex(link_header->parent_hash));
-    if (attachment_link.value()->preverified && attachment_link.value()->next.size() > 0)
+    if (attachment_link.value()->preverified && !attachment_link.value()->next.empty())
```

Comment:
I disagree. I read this as "not empty", which is more meaningful to me that "its size is greater than 0".
Also it may be faster (see https://lemire.me/blog/2021/10/26/in-c-is-empty-faster-than-comparing-the-size-with-zero/)